amazon-kinesis-client/src/main/java/software/amazon/kinesis/leases/ShardInfo.java

```diff
@@ -18,31 +18,37 @@
 import java.util.Collections;
 import java.util.LinkedList;
 import java.util.List;
-import java.util.Optional;
 
+import lombok.AccessLevel;
 import org.apache.commons.lang3.builder.EqualsBuilder;
 import org.apache.commons.lang3.builder.HashCodeBuilder;
 
 import lombok.Getter;
 import lombok.NonNull;
 import lombok.ToString;
 import lombok.experimental.Accessors;
+import software.amazon.kinesis.common.StreamConfig;
 import software.amazon.kinesis.retrieval.kpl.ExtendedSequenceNumber;
 
 /**
  * Used to pass shard related info among different classes and as a key to the map of shard consumers.
  */
 @Getter
 @Accessors(fluent = true)
-@ToString
+@ToString(exclude = {"isMultiStreamMode", "streamIdentifierStr"})
 public class ShardInfo {
 
-    private final Optional<String> streamIdentifierSerOpt;
     private final String shardId;
     private final String concurrencyToken;
     // Sorted list of parent shardIds.
     private final List<String> parentShardIds;
     private final ExtendedSequenceNumber checkpoint;
+    private final StreamConfig streamConfig;
+
+    @Getter(AccessLevel.NONE)
```

Comment:
yes thank you, this is necessary because this is class exposes a public contract which cannot be violated and we should not expose more than what is necessary. But I do agree with lucien that we probably dont need them and get it from streamConfig when needed.